[2021-04-27 21:11:59,538] {taskinstance.py:851} INFO - Dependencies all met for <TaskInstance: docker_spark_sample.docker_spark_tester 2021-04-25T02:30:00+00:00 [queued]>
[2021-04-27 21:11:59,583] {taskinstance.py:851} INFO - Dependencies all met for <TaskInstance: docker_spark_sample.docker_spark_tester 2021-04-25T02:30:00+00:00 [queued]>
[2021-04-27 21:11:59,585] {taskinstance.py:1042} INFO - 
--------------------------------------------------------------------------------
[2021-04-27 21:11:59,586] {taskinstance.py:1043} INFO - Starting attempt 1 of 2
[2021-04-27 21:11:59,588] {taskinstance.py:1044} INFO - 
--------------------------------------------------------------------------------
[2021-04-27 21:11:59,609] {taskinstance.py:1063} INFO - Executing <Task(DockerOperator): docker_spark_tester> on 2021-04-25T02:30:00+00:00
[2021-04-27 21:11:59,617] {standard_task_runner.py:52} INFO - Started process 138 to run task
[2021-04-27 21:11:59,628] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'docker_spark_sample', 'docker_spark_tester', '2021-04-25T02:30:00+00:00', '--job-id', '39', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/dockerSparkOperator.py', '--cfg-path', '/tmp/tmp7o7acufd', '--error-file', '/tmp/tmpmrx_q9fx']
[2021-04-27 21:11:59,630] {standard_task_runner.py:77} INFO - Job 39: Subtask docker_spark_tester
[2021-04-27 21:11:59,709] {logging_mixin.py:104} INFO - Running <TaskInstance: docker_spark_sample.docker_spark_tester 2021-04-25T02:30:00+00:00 [running]> on host 9e66ec21e68a
[2021-04-27 21:11:59,774] {taskinstance.py:1257} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=airflow@example.com
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=docker_spark_sample
AIRFLOW_CTX_TASK_ID=docker_spark_tester
AIRFLOW_CTX_EXECUTION_DATE=2021-04-25T02:30:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2021-04-25T02:30:00+00:00
[2021-04-27 21:11:59,794] {docker.py:224} INFO - Starting docker container from image kadirsahan/docker_spark_submit:latest
[2021-04-27 21:12:24,694] {docker.py:263} INFO - 21/04/27 21:12:23 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2021-04-27 21:12:37,559] {docker.py:263} INFO - Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
[2021-04-27 21:12:37,968] {docker.py:263} INFO - 21/04/27 21:12:36 INFO SparkContext: Running Spark version 3.1.1
[2021-04-27 21:12:39,414] {docker.py:263} INFO - 21/04/27 21:12:37 INFO ResourceUtils: ==============================================================
[2021-04-27 21:12:39,420] {docker.py:263} INFO - 21/04/27 21:12:37 INFO ResourceUtils: No custom resources configured for spark.driver.
[2021-04-27 21:12:39,442] {docker.py:263} INFO - 21/04/27 21:12:37 INFO ResourceUtils: ==============================================================
[2021-04-27 21:12:39,482] {docker.py:263} INFO - 21/04/27 21:12:37 INFO SparkContext: Submitted application: HelloWorld
[2021-04-27 21:12:40,164] {docker.py:263} INFO - 21/04/27 21:12:38 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2021-04-27 21:12:40,400] {docker.py:263} INFO - 21/04/27 21:12:38 INFO ResourceProfile: Limiting resource is cpu
[2021-04-27 21:12:40,418] {docker.py:263} INFO - 21/04/27 21:12:38 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2021-04-27 21:12:41,695] {docker.py:263} INFO - 21/04/27 21:12:40 INFO SecurityManager: Changing view acls to: root
[2021-04-27 21:12:41,700] {docker.py:263} INFO - 21/04/27 21:12:40 INFO SecurityManager: Changing modify acls to: root
[2021-04-27 21:12:41,704] {docker.py:263} INFO - 21/04/27 21:12:40 INFO SecurityManager: Changing view acls groups to:
[2021-04-27 21:12:41,711] {docker.py:263} INFO - 21/04/27 21:12:40 INFO SecurityManager: Changing modify acls groups to:
[2021-04-27 21:12:41,734] {docker.py:263} INFO - 21/04/27 21:12:40 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(root); groups with view permissions: Set(); users  with modify permissions: Set(root); groups with modify permissions: Set()
[2021-04-27 21:13:41,852] {docker.py:263} INFO - 21/04/27 21:13:40 INFO Utils: Successfully started service 'sparkDriver' on port 39449.
[2021-04-27 21:14:08,057] {docker.py:263} INFO - 21/04/27 21:14:05 INFO SparkEnv: Registering MapOutputTracker
21/04/27 21:14:06 INFO SparkEnv: Registering BlockManagerMaster
21/04/27 21:14:06 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
21/04/27 21:14:06 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2021-04-27 21:14:08,181] {docker.py:263} INFO - 21/04/27 21:14:06 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2021-04-27 21:14:08,970] {docker.py:263} INFO - 21/04/27 21:14:07 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-38276dc7-f180-4c88-9f1b-fd5e79cc28d8
[2021-04-27 21:14:11,674] {docker.py:263} INFO - 21/04/27 21:14:10 INFO MemoryStore: MemoryStore started with capacity 366.3 MiB
[2021-04-27 21:14:56,078] {docker.py:263} INFO - 21/04/27 21:14:54 INFO SparkEnv: Registering OutputCommitCoordinator
[2021-04-27 21:15:10,025] {docker.py:263} INFO - 21/04/27 21:15:08 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2021-04-27 21:15:11,939] {docker.py:263} INFO - 21/04/27 21:15:10 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://4317e75833ee:4040
[2021-04-27 21:15:34,376] {docker.py:263} INFO - 21/04/27 21:15:32 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2021-04-27 21:15:39,904] {docker.py:263} INFO - 21/04/27 21:15:38 WARN TransportClientFactory: DNS resolution failed for spark-master:7077 took 5030 ms
[2021-04-27 21:15:54,336] {docker.py:263} INFO - 21/04/27 21:15:52 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2021-04-27 21:15:55,337] {docker.py:263} INFO - 21/04/27 21:15:53 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
org.apache.spark.SparkException: Exception thrown in awaitResult: 
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:301)
	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:101)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:109)
	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:107)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Failed to connect to spark-master:7077
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:287)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:218)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
	... 4 more
Caused by: java.net.UnknownHostException: spark-master: Try again
	at java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:930)
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1325)
	at java.net.InetAddress.getAllByName0(InetAddress.java:1278)
	at java.net.InetAddress.getAllByName(InetAddress.java:1194)
	at java.net.InetAddress.getAllByName(InetAddress.java:1128)
	at java.net.InetAddress.getByName(InetAddress.java:1078)
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
	at java.security.AccessController.doPrivileged(Native Method)
	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:206)
	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:180)
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:166)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490)
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615)
	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:604)
	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:984)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:504)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:417)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:474)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 more
[2021-04-27 21:16:14,328] {docker.py:263} INFO - 21/04/27 21:16:12 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2021-04-27 21:16:19,307] {docker.py:263} INFO - 21/04/27 21:16:17 WARN TransportClientFactory: DNS resolution failed for spark-master:7077 took 5006 ms
[2021-04-27 21:16:19,323] {docker.py:263} INFO - 21/04/27 21:16:17 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
org.apache.spark.SparkException: Exception thrown in awaitResult: 
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:301)
	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:101)
	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:109)
	at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:107)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Failed to connect to spark-master:7077
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:287)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:218)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
	at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
	... 4 more
Caused by: java.net.UnknownHostException: spark-master
	at java.net.InetAddress.getAllByName0(InetAddress.java:1282)
	at java.net.InetAddress.getAllByName(InetAddress.java:1194)
	at java.net.InetAddress.getAllByName(InetAddress.java:1128)
	at java.net.InetAddress.getByName(InetAddress.java:1078)
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
	at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
	at java.security.AccessController.doPrivileged(Native Method)
	at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
	at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
	at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
	at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
	at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:206)
	at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:180)
	at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:166)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:577)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:551)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:490)
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:615)
	at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:604)
	at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
	at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:984)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:504)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:417)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:474)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 more
[2021-04-27 21:16:34,319] {docker.py:263} INFO - 21/04/27 21:16:32 ERROR StandaloneSchedulerBackend: Application has been killed. Reason: All masters are unresponsive! Giving up.
[2021-04-27 21:16:34,323] {docker.py:263} INFO - 21/04/27 21:16:32 WARN StandaloneSchedulerBackend: Application ID is not initialized yet.
[2021-04-27 21:16:34,492] {docker.py:263} INFO - 21/04/27 21:16:33 INFO SparkUI: Stopped Spark web UI at http://4317e75833ee:4040
[2021-04-27 21:16:34,496] {docker.py:263} INFO - 21/04/27 21:16:33 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 36417.
[2021-04-27 21:16:34,498] {docker.py:263} INFO - 21/04/27 21:16:33 INFO NettyBlockTransferService: Server created on 4317e75833ee:36417
[2021-04-27 21:16:34,535] {docker.py:263} INFO - 21/04/27 21:16:33 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2021-04-27 21:16:34,648] {docker.py:263} INFO - 21/04/27 21:16:33 INFO StandaloneSchedulerBackend: Shutting down all executors
[2021-04-27 21:16:34,696] {docker.py:263} INFO - 21/04/27 21:16:33 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
[2021-04-27 21:16:34,731] {docker.py:263} INFO - 21/04/27 21:16:33 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 4317e75833ee, 36417, None)
[2021-04-27 21:16:34,762] {docker.py:263} INFO - 21/04/27 21:16:33 WARN StandaloneAppClient$ClientEndpoint: Drop UnregisterApplication(null) because has not yet connected to master
[2021-04-27 21:16:34,840] {docker.py:263} INFO - 21/04/27 21:16:33 INFO BlockManagerMasterEndpoint: Registering block manager 4317e75833ee:36417 with 366.3 MiB RAM, BlockManagerId(driver, 4317e75833ee, 36417, None)
[2021-04-27 21:16:34,866] {docker.py:263} INFO - 21/04/27 21:16:33 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2021-04-27 21:16:34,871] {docker.py:263} INFO - 21/04/27 21:16:33 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 4317e75833ee, 36417, None)
[2021-04-27 21:16:34,884] {docker.py:263} INFO - 21/04/27 21:16:33 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 4317e75833ee, 36417, None)
[2021-04-27 21:16:34,952] {docker.py:263} INFO - 21/04/27 21:16:33 INFO MemoryStore: MemoryStore cleared
[2021-04-27 21:16:34,953] {docker.py:263} INFO - 21/04/27 21:16:33 INFO BlockManager: BlockManager stopped
[2021-04-27 21:16:35,155] {docker.py:263} INFO - 21/04/27 21:16:33 INFO BlockManagerMaster: BlockManagerMaster stopped
[2021-04-27 21:16:35,183] {docker.py:263} INFO - 21/04/27 21:16:33 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2021-04-27 21:16:35,221] {docker.py:263} INFO - 21/04/27 21:16:33 INFO SparkContext: Successfully stopped SparkContext
[2021-04-27 21:16:37,870] {taskinstance.py:1455} ERROR - docker container failed: {'Error': None, 'StatusCode': 137}
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1112, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1285, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1315, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/docker/operators/docker.py", line 307, in execute
    return self._run_image()
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/docker/operators/docker.py", line 269, in _run_image
    raise AirflowException('docker container failed: ' + repr(result))
airflow.exceptions.AirflowException: docker container failed: {'Error': None, 'StatusCode': 137}
[2021-04-27 21:16:37,873] {taskinstance.py:1503} INFO - Marking task as UP_FOR_RETRY. dag_id=docker_spark_sample, task_id=docker_spark_tester, execution_date=20210425T023000, start_date=20210427T211159, end_date=20210427T211637
[2021-04-27 21:16:37,926] {local_task_job.py:146} INFO - Task exited with return code 1
